src/Catalyst.Core.Lib.Tests/IntegrationTests/FileSystem/FileStoreTest.cs

```diff
@@ -0,0 +1,199 @@
+#region LICENSE
+
+/**
+* Copyright (c) 2019 Catalyst Network
+*
+* This file is part of Catalyst.Node <https://github.com/catalyst-network/Catalyst.Node>
+*
+* Catalyst.Node is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 2 of the License, or
+* (at your option) any later version.
+*
+* Catalyst.Node is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Catalyst.Core.Lib.FileSystem;
+using Catalyst.TestUtils;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Catalyst.Core.Lib.Tests.UnitTests
+{
+    public class FileStoreTest : FileSystemBasedTest
+    {
+        public FileStoreTest(ITestOutputHelper output) : base(output) { }
+        
+        class Entity
+        {
+            public int Number;
+            public string Value;
+        }
+
+        Entity a = new Entity {Number = 1, Value = "a"};
+        Entity b = new Entity {Number = 2, Value = "b"};
+
+        FileStore<int, Entity> Store
+        {
+            get
+            {
+                var folder = Path.Combine(FileSystem.GetCatalystDataDir().FullName, "test-filestore");
+                if (!Directory.Exists(folder))
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)